libbeat/processors/add_cloud_metadata/add_cloud_metadata.go

```diff
@@ -358,9 +358,15 @@ func (p *addCloudMetadata) Run(event *beat.Event) (*beat.Event, error) {
 		return event, nil
 	}
 
-	// This overwrites the meta.cloud if it exists. But the cloud key should be
-	// reserved for this processor so this should happen.
-	_, err := event.PutValue("cloud", meta)
+	// If cloud key exists in event already, this processor will not overwrite the cloud fields.
+	// For example aws module writes cloud.instance.* to events already, add_cloud_metadata should not overwrite these
+	// fields again.
+	cloudValue, err := event.GetValue("cloud")
```

Comment:
You only care about whether or not `cloud` is non-nil and since `GetValue` will never return both an error and non-nil value it's fine to do:
 
```
	cloudValue, _ := event.GetValue("cloud")
	if cloudValue != nil {
	    return event, nil
	}
```

Reply:
Good point! I will change to only check cloudValue then! Thank you for your review!